src/Voron/Data/Lookups/Lookup.cs

```diff
@@ -408,7 +408,9 @@ private void RemoveEntryFromPage(ref CursorState state, ref TLookupKey key, int
         Debug.Assert(state.Header->FreeSpace <= Constants.Storage.PageSize - PageHeader.SizeOf);
         Debug.Assert(k == key.ToLong(), "k == key.ToLong()");
         
-        key.OnKeyRemoval(this);
+        if (state.Header->IsBranch || isExplicitRemove)
+            key.OnKeyRemoval(this);
```

Comment:
What we want to achieve here is that we don't want to trigger `OnKeyRemoval()` for internal operations on leafs e.g. moving entries from one leaf to another during tree rebalancing on deletions. This way we won't decrement the term reference count.